src/bun.js/node/node_perf_hooks_histogram_binding.zig

```diff
@@ -0,0 +1,234 @@
+            return globalThis.toJS(this.hdrHist.total_count, .temporary);
+        }
+    };
+    pub const count = @as(PropertyGetter, count_fn.callback);
+    pub const countBigInt = getterAsFn(count_fn.callback);
+
+    pub const mean = @as(
+        PropertyGetter,
+        struct {
+            pub fn callback(this: *This, globalThis: *JSC.JSGlobalObject) callconv(.C) JSValue {
+                if (this.hdrHist.mean()) |m| {
+                    return globalThis.toJS(m, .temporary);
+                }
+                return globalThis.toJS(std.math.nan(f64), .temporary);
+            }
+        }.callback,
+    );
+
+    pub const stddev = @as(
+        PropertyGetter,
+        struct {
+            pub fn callback(this: *This, globalThis: *JSC.JSGlobalObject) callconv(.C) JSValue {
+                if (this.hdrHist.stddev()) |sd| {
+                    return globalThis.toJS(sd, .temporary);
+                }
+                return globalThis.toJS(std.math.nan(f64), .temporary);
+            }
+        }.callback,
+    );
+
+    pub const percentile = struct {
+        pub fn callback(this: *This, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+            const args = callframe.arguments(1).slice();
+            if (args.len != 1) {
+                globalThis.throwInvalidArguments("Expected 1 argument", .{});
+                return .zero;
+            }
+            const percent = args[0].getNumber() orelse {
+                globalThis.throwInvalidArguments("Expected a number", .{});
+                return .zero;
+            };
+            const value = this.hdrHist.value_at_percentile(percent) orelse return .undefined;
+            return globalThis.toJS(value, .temporary);
+        }
+    }.callback;
+    pub const percentileBigInt = percentile;
+
+    extern fn Bun__createMapFromDoubleUint64TupleArray(
+        globalObject: *JSC.JSGlobalObject,
+        doubles: [*]const f64,
+        length: usize,
+    ) JSC.JSValue;
+
+    const percentiles_fn = struct {
+        pub fn callback(this: *This, globalObject: *JSC.JSGlobalObject) callconv(.C) JSValue {
+
+            // 2 arrays with percent and value
+            // make a cpp version of this file, extern C function. accepts array, length, creates search JSMap:: (search)
+
+            // first get 100th percentile, and loop 0, 50, 75, 82.5, ... until we find the highest percentile
+            const maxPercentileValue = this.hdrHist.value_at_percentile(100) orelse return .undefined;
+            var percent: f64 = 0;
+            var stack_allocator = std.heap.stackFallback(4096, bun.default_allocator);
+            var doubles = std.ArrayList(f64).init(stack_allocator.get());
+            defer doubles.deinit();
+
+            while (true) {
+                if (this.hdrHist.value_at_percentile(percent)) |val| {
+                    doubles.appendSlice(&.{ percent, @bitCast(val) }) catch |err| {
```

Comment:
This is not how [alignment](https://en.wikipedia.org/wiki/Data_structure_alignment) works. and even if it was, this would not double memory usage.

`@sizeOf(DoubleToIntMapKV)` is `16`
`@alignOf(DoubleToIntMapKV)` is `8`

But even if the alignment was 16, the memory layout of a slice would still have no padding, it just would be have 16 bit alignment.

Reply:
edit: i'm wrong

Reply:
i'll do extern structs then!

Reply:
done